In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import re
%matplotlib inline

In [3]:
df = pd.read_csv('C:/Users/Phillip/Pictures/Box Sync/My Python Programs/Data Science/cleanchallenge.csv', encoding='latin1')

In [5]:
print(df.head(5))
#df['COST (£) charged to Wellcome (inc VAT when charged)']

              PMID/PMCID Publisher           Journal title  \
0                    NaN       CUP  Psychological Medicine   
1             PMC3679557       ACS       Biomacromolecules   
2  23043264  PMC3506128        ACS              J Med Chem   
3    23438330 PMC3646402       ACS              J Med Chem   
4   23438216 PMC3601604        ACS              J Org Chem   

                                       Article title  \
0  Reduced parahippocampal cortical thickness in ...   
1  Structural characterization of a Model Gram-ne...   
2  Fumaroylamino-4,5-epoxymorphinans and related ...   
3  Orvinols with mixed kappa/mu opioid receptor a...   
4  Regioselective opening of myo-inositol orthoes...   

  COST (£) charged to Wellcome (inc VAT when charged)  
0                                              £0.00   
1                                           £2381.04   
2                                            £642.56   
3                                            £669.64   
4         

# Cleaning Tasks

Going to need to go over publisher names and get everything to match up, for example, ACS/American Chemical Society to the same referrence.

Journal Titles J Med Chem/Journal of Medicinal Chemistry title clean up

Remove pound/sterling symbol

In [6]:
df['Journal title'] = df['Journal title'].replace(['J Med Chem', 'Journal of Medicinal Chemistry', 'J Org Chem'], 'JMC')
df['Journal title'] = df['Journal title'].replace(['ACS Chemical Biology', 'ACS Chemical Neuroscience', 'American Chemical Society', 'ACS NANA'], 'ACS')
df['Journal title'] = df['Journal title'].replace(['Journal of Biochemistry', 'Biochemistry', 'Bioconjugate Chemistry', ], 'JoB')
df['Journal title'] = df['Journal title'].replace(['OUP', 'Oxford', 'Oxford Journals', ], 'Oxford University Press')


In [7]:
print(df['Journal title'])

0                                  Psychological Medicine
1                                       Biomacromolecules
2                                                     JMC
3                                                     JMC
4                                                     JMC
5                                                     JMC
6                            Journal of Proteome Research
7                                               Mol Pharm
8                                                     ACS
9                                                     ACS
10           Journal of Chemical Information and Modeling
11                                                    JMC
12                                                    JMC
13                                                    JoB
14                                                    JMC
15                                       Gastroenterology
16                        Journal of Biological Chemistry
17            

In [8]:
pub_counts = df.groupby('Publisher')['Publisher'].count()

pub_counts.sort_values(ascending=False)
    

Publisher
Elsevier                                                                             387
Public Library of Science                                                            278
Wiley                                                                                136
Springer                                                                              81
Oxford University Press                                                               77
OUP                                                                                   56
Wiley-Blackwell                                                                       56
ASBMB                                                                                 46
Nature Publishing Group                                                               45
BioMed Central                                                                        40
BMC                                                                                   26
Nature     

# Determine 5 most common journals and their corresponding total articles

# Top 5 Publishers
1. Elsevier 387 articles
2. Public Library of Science 287 articles
3. Wiley 136 articles
4. Oxford University Press 133 articles 
5. Springer 81 articles


# Calculate mean, median, std dev cost per article per journal

In [20]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2127 entries, 0 to 2126
Data columns (total 5 columns):
PMID/PMCID                                             1928 non-null object
Publisher                                              2127 non-null object
Journal title                                          2126 non-null object
Article title                                          2127 non-null object
COST (£) charged to Wellcome (inc VAT when charged)    2127 non-null object
dtypes: object(5)
memory usage: 83.2+ KB


In [9]:
# df.rename(columns={'COST (£) charged to Wellcome (inc VAT when charged)': 'Cost'})

nosign = df['COST (£) charged to Wellcome (inc VAT when charged)'].replace('£', '', regex=True)

In [13]:
# need to convert numeric???